src/ray/raylet/scheduling/cluster_resource_data.cc

```diff
@@ -44,10 +44,12 @@ ResourceRequest ResourceMapToResourceRequest(
 /// \request Conversion result to a ResourceRequest data structure.
 NodeResources ResourceMapToNodeResources(
     const absl::flat_hash_map<std::string, double> &resource_map_total,
-    const absl::flat_hash_map<std::string, double> &resource_map_available) {
+    const absl::flat_hash_map<std::string, double> &resource_map_available,
+    const absl::flat_hash_map<std::string, std::string> &node_labels) {
   NodeResources node_resources;
   node_resources.total = ResourceMapToResourceRequest(resource_map_total, false);
   node_resources.available = ResourceMapToResourceRequest(resource_map_available, false);
+  node_resources.labels = node_labels;
```

Comment:
After we have labels, the current naming (NodeResources) is no longer that accurate. We should probably have something like

```
class Node {
   NodeResources resources;
  map<string, string> labels;
}
```

Reply:
1. Labels can also be regarded as a resource of node. So it is very suitable to put it in NodeResources, and it reduces a lot of code changes.
2. NodeResources do not need to be computable. The ResourceRequest in NodeResources is Computable. Adding labels to NodeResources does not affect the original available/tatal, etc.
![image](https://github.com/ray-project/ray/assets/11072802/b9dda790-599a-4cdc-8664-d78308987dff)

Reply:
Cool! Got the design.